### DISCUSSION TITLE
Is the library SQL injection safe?

### DISCUSSION THREAD

**Discussion by kishaningithub** [aws/amazon-redshift-python-driver#77] [Q&A] [answered]
If i pass in the parameters of a query directly from an input source.. are they sanitized to prevent SQL injection?

Or should something like [jinjasql](https://github.com/sripathikrishnan/jinjasql) should be used to sanitize the SQL params?

**Comment by Brooke-white** [comment, marked as answer]
redshift_connector uses parameterized statements by default, as a protector against SQL injection.

```python
cursor.execute("select * from my_table where some_col = %s", (some_value,))
```

Using parameterized statements is safer than using approaches such as string concatenation, shown below

```python
cursor.execute("select * from my_table where some_col = {}".format("'hello world'")
```

Please let me know if there is anything else I can clarify.
